src/bindgen.rs

```diff
@@ -42,33 +73,88 @@ pub fn wasm_bindgen_build(
+        );
+        let dts_arg = if disable_dts == false {
+            "--typescript"
+        } else {
+            "--no-typescript"
+        };
+        let target_arg = match target {
+            "nodejs" => "--nodejs",
+            _ => "--browser",
+        };
+        let bindgen_path = Path::new(&wasm_bindgen_path_str);
+        let output = Command::new(bindgen_path)
+            .current_dir(path)
+            .arg(&wasm_path)
+            .arg("--out-dir")
+            .arg("./pkg")
+            .arg(dts_arg)
+            .arg(target_arg)
+            .output()?;
+        if !output.status.success() {
+            let s = String::from_utf8_lossy(&output.stderr);
+            Error::cli("wasm-bindgen failed to execute properly", s)
+        } else {
+            Ok(())
+        }
+    } else {
+        Error::crate_config("Could not find `wasm-bindgen`")
+    }
+}
 
-    let output = Command::new("wasm-bindgen")
-        .current_dir(path)
-        .arg(&wasm_path)
-        .arg("--out-dir")
-        .arg("./pkg")
-        .arg(dts_arg)
-        .arg(target_arg)
-        .output()?;
-    if !output.status.success() {
+/// Check if the `wasm-bindgen` dependency is locally satisfied.
+fn wasm_bindgen_version_check(bindgen_path: &PathBuf, dep_version: &str) -> Result<bool, Error> {
+    let wasm_bindgen = Path::new(bindgen_path);
+    let output = Command::new(wasm_bindgen).arg("--version").output()?;
+    if output.status.success() {
+        let s = String::from_utf8_lossy(&output.stdout);
+        let installed_version = s.trim(); // FIXUP: Trim this more thoroughly?
+        Ok(installed_version == dep_version)
+    } else {
+        // FIXUP: This error message can be improved.
+        let message = String::from("Could not find version of local wasm-bindgen");
         let s = String::from_utf8_lossy(&output.stderr);
-        Error::cli("wasm-bindgen failed to execute properly", s)
+        Err(Error::Cli {
+            message,
+            stderr: s.to_string(),
+        })
+    }
+}
+
+/// Return a string to `wasm-bindgen`, if it exists for the given mode.
+fn wasm_bindgen_path_str(crate_path: &Path) -> Option<PathBuf> {
+    if let res @ Some(_) = local_wasm_bindgen_path_str(crate_path) {
+        res
+    } else if let res @ Some(_) = global_wasm_bindgen_path_str() {
+        res
     } else {
-        Ok(())
+        None
+    }
+}
+
+/// Return a string containing the path to the local `wasm-bindgen`, if it exists.
+fn local_wasm_bindgen_path_str(crate_path: &Path) -> Option<PathBuf> {
+    let mut path = crate_path.to_path_buf();
+    path.push("bin");
+    path.push("wasm-bindgen");
+    if path.is_file() {
+        Some(path)
+    } else {
+        None
+    }
+}
+
+/// Return a string containing the path to the global `wasm-bindgen`, if it exists.
+fn global_wasm_bindgen_path_str() -> Option<PathBuf> {
+    if let Ok(path) = which("wasm-bindgen") {
+        Some(path)
+    } else {
+        None
```

Comment:
Other details I know I need to take care of: These functions can be collapsed into a single function, and should be named `wasm_bindgen_path` rather than being suffixed `path_str`, since these return `PathBuf` values rather than strings now.